tests/test_appliancesmanager.py

```diff
@@ -121,3 +121,100 @@ async def test_fetch_appliances_returns_true_if_either_method_returns_true(
 
     assert result == bool(owned_response or shared_response)
     await http_client_mock.close_session()
+
+
+@pytest.mark.parametrize(
+    "data_model_key",
+    [
+        "DDM_COOKING_BIO_SELF_CLEAN_TOURMALINE_V2",
+        "COOKING_MINERVA",
+        "COOKING_VSI",
+        "COOKING_U2",
+    ],
+)
+async def test_add_appliance(data_model_key, http_client_mock: AiohttpClientMocker):
+    appliance = {
+        "DATA_MODEL_KEY": f"{data_model_key}",
+        "CATEGORY_NAME": "Cooking",
+        "MODEL_NO": "000000000000",
+        "REPLENISHMENT_DEVICE_MODEL": None,
+        "IMAGE_PATH": None,
+        "APPLIANCE_ID": 95842,
+        "APPLIANCE_MASTER_ID": 491,
+        "MODEL_SKU_ID": None,
+        "CREATED_AT": 1715804778000,
+        "UPDATED_AT": 1715804778000,
+        "APPLIANCE_NAME": "Oven",
+        "SAID": "0000000000000",
+        "NEST_AWAY": 0,
+        "CYCLE_HANDOFF": 0,
+        "NEST_THERMOSTAT_ID": 0,
+        "THERMOSTAT_INFLUENCE_THRESHOLD": None,
+        "THERMOSTAT_DESIRED_OFFSET": None,
+        "THERMOSTAT_OFFSET_NEEDED": None,
+        "DELETE_FLAG": 0,
+        "DISPLAY_POSITION": None,
+        "SERIAL": "000000000000",
+        "LOCATION_ID": 260998,
+        "MACHINE_ID": None,
+        "MACHINE_POSITION": 0,
+        "ISVOICEDEFAULT": 1,
+        "DEVICE_ID": "aaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa",
+        "IS_ENROLLED": None,
+        "STATUS": "CLAIMED",
+        "APPLIANCE_TYPE_ID": None,
+        "MACHINE_STATUS": None,
+        "APPLIANCE_MODE": 2,
+    }
+    appliance_data = {
+        "CATEGORY": "Cooking",
+        "DATA_MODEL": f"{data_model_key}",
+        "MODEL_NUMBER": "000000000000",
+        "NAME": "Oven",
+        "SAID": "0000000000000",
+        "SERIAL_NUMBER": "000000000000",
+    }
+    auth = Auth(BACKEND_SELECTOR_MOCK, "email", "secretpass", http_client_mock.session)
+    am = AppliancesManager(BACKEND_SELECTOR_MOCK, auth, http_client_mock.session)
+    am._add_appliance(appliance)
```

Comment:
I would prefer to avoid using internal class methods in the tests. 
Can we switch this to use only external stuff (in this case probably `http_client_mock` and `am.fetch_appliances()`

Reply:
Sure, I'll do that, but I focused on unit testing this specific method only ;)

Reply:
just extended actual unittest about oven